### ISSUE TITLE
CSS for tier prices block still visible when there are no tier prices

### ISSUE THREAD

**Issue by posidonius** [magento/magento2#29194]
#26081 
# Preconditions (*)
Magento 2.4

### Steps to reproduce (*)
1. Create a configurable product and associate 1 simple product with no tier pricing and 1 simple product with tier pricing
2. Style the tier prices block with a background or border (for example `border: aqua; background: #99cfff;`)
3. Toggle between the 2 simple products. For the product with no tier pricing, the content disappears but the CSS styling remains (e.g. in Magento Blank theme, there is an empty box with a gray background).

![Screenshot_46](https://user-images.githubusercontent.com/60198022/88168322-e881da80-cc22-11ea-8464-e4a903566656.png)
![Screenshot_47](https://user-images.githubusercontent.com/60198022/88168293-e1f36300-cc22-11ea-8ca9-f62efd66d81c.png)

### Expected result (*)
1. The tier price block, including associated CSS, should disappear from the frontend when choosing a simple product that doesn't have tier prices.

### Actual result (*)
1. The tier price block, including CSS styling such as background or border, is visible even when there are no tier prices to display.
![Screenshot_47](https://user-images.githubusercontent.com/60198022/88168293-e1f36300-cc22-11ea-8ca9-f62efd66d81c.png)

**Additional information**
Even when there are no tier prices, any CSS applied to tier prices block are visible on the frontend.

A simple edit to _displayTierPriceBlock function in configurable.js can fix this problem.
**Solution:**

in _displayTierPriceBlock function of configurable.js, change the following if statement:

```
            if (typeof optionId != 'undefined' &&
                this.options.spConfig.optionPrices[optionId].tierPrices != [] // eslint-disable-line eqeqeq
            ) 
```

to the following, which includes a check for tier prices array with 0 length:

```
            if (typeof optionId != 'undefined' &&
                this.options.spConfig.optionPrices[optionId].tierPrices != [] && // eslint-disable-line eqeqeq
                this.options.spConfig.optionPrices[optionId].tierPrices.length > 0
            ) 
```

---

- [S4 ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Lima 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-36077` were created

**Issue Available**: @engcom-Lima, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Alfa** [comment]
The issue is reproducible on fresh 2.4-develop

![Peek 2020-07-23 15-51](https://user-images.githubusercontent.com/51679138/88288487-84cbe000-ccfc-11ea-9912-3ff91ff598d9.gif)

**Comment by srsathish92** [comment]
@magento I am working on this

**Comment by magento-engcom-team** [comment]
Hi @posidonius. Thank you for your report. 
The issue has been fixed in magento/magento2#29353 by @srsathish92 in 2.4-develop branch
Related commit(s):
 - [3d7298e3d10a77b5321a152bfec9eeea634b079d](https://github.com/magento/magento2/commit/3d7298e3d10a77b5321a152bfec9eeea634b079d)

The fix will be available with the upcoming 2.4.2 release.
